leetcode/add_two_numbers/step3.py

```diff
@@ -0,0 +1,20 @@
+from typing import Optional
+
+class Solution:
+    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]): Optional[ListNode]:
+        dummy = ListNode()
+        node = dummy
+        carry = 0
+        while l1 is not None or l2 is not None or carry != 0:
```

Comment:
細かい好みですが、変数宣言部分とwhile文は改行で1行あけると読みやすいなと思いました。
```suggestion

        while l1 is not None or l2 is not None or carry != 0:
```